lib/codegen/src/context.rs

```diff
@@ -123,6 +124,10 @@ impl Context {
         let _tt = timing::compile();
         self.verify_if(isa)?;
 
+        if isa.flags().enable_constant_folding() {
+            self.fold_constants(isa)?;
+        }
```

Comment:
A concern that I have is that this may increase the size of the compiler for users that aren't using this. I don't know how big the constant folding pass will get, but if we're going to be getting into optimizing unoptimized code, we may be adding other things besides.

Since this is running at the beginning of `compile`, what if we moved it out into a separate top-level `optimize` function on `Context`? Users that know they have unoptimized code could call `optimize` before `compile`, and for users that don't call `optimize`, it could get DCE'd.

Reply:
Okay, that sounds good to me. That would entail removing the `constant_folding` setting as well, right?

Reply:
Yes, that's right.